src/jit/codegencommon.cpp

```diff
@@ -647,7 +647,7 @@ regMaskTP Compiler::compHelperCallKillSet(CorInfoHelpFunc helper)
 #if defined(_TARGET_AMD64_)
             return RBM_RSI | RBM_RDI | RBM_CALLEE_TRASH;
 #elif defined(_TARGET_ARM64_)
-            return RBM_CALLEE_TRASH_NOGC;
+            return RBM_R12 | RBM_R15;
```

Comment:
I would have preferred a simpler change to just use compHelperCallKillSet, but it is used in other places.

The most problematic was here

https://github.com/dotnet/coreclr/blob/master/src/jit/codegenarm64.cpp#L6144 

I think the change is functionally correct as it fixed several GC failures.  There may be CQ issues.  Let me know if you want changes.